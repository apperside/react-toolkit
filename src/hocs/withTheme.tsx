import React from 'react';
import darkTheme, { customTheme, ThemeProvider } from '../ui/theme';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
    const ComponentWithTheme = (props: P) => {
        return (
            <ThemeProvider theme={customTheme as any}>
                <WrappedComponent {...props} />
            </ThemeProvider>
        );
    };
    return ComponentWithTheme;
}
export default withTheme;
