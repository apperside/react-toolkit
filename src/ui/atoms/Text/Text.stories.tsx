import React from 'react';
import { Story, Meta } from '@storybook/react';

import Text from './Text';
import { TextProps } from "./Text.styled";

export default {
    title: 'Components/Text',
    component: Text,
    argTypes: {
        backgroundColor: { control: 'color' },
        size: {
            control: {
                type: 'select',
                options: ['small', 'medium', 'large'],
            },
        },
        borderRadius: {
            control: {
                type: 'select',
                options: [2, 4, 8],
            },
        },
    },
} as Meta;

const Template: Story<TextProps> = (args) => <Text {...args} >Hello world</Text>;

export const Primary = Template.bind({});
Primary.args = {
    variant: 'regular',
    // label: 'Text',
};

export const Secondary = Template.bind({});
Secondary.args = {
    variant: 'bold',
    // label: 'Text',
};

export const Danger = Template.bind({});
Danger.args = {
    variant: 'medium',
    // label: 'Text',
};

export const Large = Template.bind({});
Large.args = {
    size: 'black',
    // label: 'Text',
};
