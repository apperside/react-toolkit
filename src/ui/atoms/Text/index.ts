export * from "./Text"
export * from "./Text.styled"
export * from "./Text.theme"