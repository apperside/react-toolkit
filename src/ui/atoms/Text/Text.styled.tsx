import { CursorProps, PropsWithVariant } from "../../theme";
import styled from 'styled-components';
import {
    BackgroundImageProps, border, BorderProps, borderRadius, color,
    ColorProps, flexbox, FlexboxProps, FontFamilyProps, fontSize, layout, LayoutProps, padding,
    position, space, SpaceProps, textStyle, TextStyleProps as SSTextStyleProps, typography,
    TypographyProps as SSTypographyProps, variant, WidthProps
} from 'styled-system';
import { TextVariants } from "./Text.theme";

export type TextStyleProps =
    & SSTypographyProps
    & SpaceProps
    & WidthProps
    & ColorProps
    & BackgroundImageProps
    & SSTextStyleProps
    & LayoutProps
    & FlexboxProps
    & FontFamilyProps
    & BorderProps
    & CursorProps

export interface TextProps extends TextStyleProps, PropsWithVariant<keyof TextVariants> {
    // children: React.ReactNode;
    // label?: string
    // onClick?: () => void
}

export const ThemedText = styled.span<TextProps>(
    {
        // appearance: 'text',
        border: 0,
        outline: 0,
    },
    borderRadius,
    // textSize,
    textStyle,
    color,
    space,
    fontSize,
    color,
    space,
    layout,
    padding,
    flexbox,
    border,
    typography,
    position,
    props => {
        return variant({
            variants: props.theme.text.variants as any
        })
    },

);
