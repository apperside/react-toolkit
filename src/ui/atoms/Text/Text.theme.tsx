import { VariantProps } from "../../theme";
import { TextStyleProps } from "./Text.styled";


// type BaseTextVariants = { [key: string]: VariantProps<TextProps> }

export interface TextVariants {
  regular: VariantProps<TextStyleProps>;
  bold: VariantProps<TextStyleProps>;
  medium: VariantProps<TextStyleProps>;
  black: VariantProps<TextStyleProps>;
}

/**
 * The new type for a themed component with variants
 * @param Props type of component's props
 * @param Variant the type defining the component's variants
 * 
 * EG: type MyComponentVariants = "primary" | "secondary"
 */
// export type ThemedComponentWithVariants<Props, Variants extends string> =
//   Props & { variants: VariantsProps<Props, Variants> }

export type TextTheme = TextStyleProps & { variants: Partial<TextVariants> }
// interface Variants<T>={[key:keyof:T]}
export const textTheme: TextTheme = {
  variants: {
    regular: {
      fontFamily: "regular",
      // backgroundColor: "yellow",
    },
    bold: {
      fontFamily: "bold"
    },
    medium: {
      fontFamily: "medium"
    },
    black: {
      fontFamily: "black"
    }
  }
};