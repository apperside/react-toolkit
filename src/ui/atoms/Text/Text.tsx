import React from 'react';
import { TextProps, ThemedText } from './Text.styled';

export const Text: React.FC<TextProps> = ({
    variant = "regular",
    ...props
}) => {
    return (
        <ThemedText variant={variant} {...props}  >
            {props.children}
        </ThemedText >
    )
};

export default Text;
