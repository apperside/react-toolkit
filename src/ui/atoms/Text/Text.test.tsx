import { render, screen } from '@testing-library/react';
import React from 'react';
import withTheme from "src/hocs/withTheme";
import Text from './Text';

const ThemedText = withTheme(Text);
describe('Text', () => {
    test('renders a default text', () => {
        const { baseElement } = render(<ThemedText >Hello world</ThemedText>);

        expect(screen.getByText('Hello world')).toBeInTheDocument();
        // expect(screen.getByText('Click me')).toHaveStyle({
        //     backgroundColor: '#025D92',
        //     color: 'white',
        // });
        expect(baseElement).toMatchSnapshot();
    });
    // test('renders a primary button', () => {
    //     const { baseElement } = render(<ThemedText variant="primary" label={'Click me'} />);

    //     expect(screen.getByText('Click me')).toHaveStyle({
    //         backgroundColor: '#025D92',
    //         color: 'white',
    //     });
    //     expect(baseElement).toMatchSnapshot();
    // });
    // test('handles onClick', () => {
    //     const mockOnClick = jest.fn();
    //     const { baseElement } = render(<ThemedText label={'Click Me'} onClick={mockOnClick} />);
    //     userEvent.click(screen.getByText('Click Me'));

    //     expect(mockOnClick).toHaveBeenCalledTimes(1);
    //     expect(baseElement).toMatchSnapshot();
    // });
});
