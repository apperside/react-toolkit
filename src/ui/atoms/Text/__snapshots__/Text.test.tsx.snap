// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders a default text 1`] = `
<body>
  <div>
     
    <span
      class="sc-bdnxRM gOGbfd"
    >
      Hello world
    </span>
     
  </div>
</body>
`;
